src/tests.rs

```diff
@@ -0,0 +1,192 @@
+            {
+                continue;
+            }
+        }
+        println!("{} -> ", path.file_name().unwrap().to_str().unwrap());
+        let file = File::open(path).unwrap();
+        let jd = &mut serde_json::Deserializer::from_reader(file.try_clone().unwrap());
+        let _: PresentationDefinitionTest = serde_path_to_error::deserialize(jd)
+            .map_err(|e| e.path().to_string())
+            .unwrap();
+        println!("✅")
+    }
+}
+
+#[derive(Deserialize)]
+pub struct PresentationSubmissionTest {
+    #[serde(alias = "presentation_submission")]
+    _ps: PresentationSubmission,
+}
+
+#[test]
+fn presentation_submission_suite() {
+    let paths = fs::read_dir("tests/presentation-exchange/test/presentation-submission").unwrap();
+    for path in paths {
+        let path = path.unwrap().path();
+        if let Some(ext) = path.extension() {
+            if ext != OsStr::new("json")
+                || [
+                    "appendix_DIDComm_example.json",
+                    "appendix_CHAPI_example.json",
+                ]
+                .contains(&path.file_name().unwrap().to_str().unwrap())
+            {
+                continue;
+            }
+        }
+        println!("{} -> ", path.file_name().unwrap().to_str().unwrap());
+        let file = File::open(path).unwrap();
+        let jd = &mut serde_json::Deserializer::from_reader(file.try_clone().unwrap());
+        let _: PresentationSubmissionTest = serde_path_to_error::deserialize(jd)
+            .map_err(|e| e.path().to_string())
+            .unwrap();
+        println!("✅")
+    }
+}
+
+#[derive(Deserialize)]
+pub struct SubmissionRequirementsTest {
+    #[serde(alias = "submission_requirements")]
+    _sr: Vec<SubmissionRequirement>,
+}
+
+#[test]
+fn submission_requirements_suite() {
+    let paths = fs::read_dir("tests/presentation-exchange/test/submission-requirements").unwrap();
+    for path in paths {
+        let path = path.unwrap().path();
+        if let Some(ext) = path.extension() {
+            if ext != OsStr::new("json")
+                || ["schema.json"].contains(&path.file_name().unwrap().to_str().unwrap())
+            {
+                continue;
+            }
+        }
+        print!("{} -> ", path.file_name().unwrap().to_str().unwrap());
+        let file = File::open(path).unwrap();
+        let jd = &mut serde_json::Deserializer::from_reader(file.try_clone().unwrap());
+        let _: SubmissionRequirementsTest = serde_path_to_error::deserialize(jd)
+            .map_err(|e| e.path().to_string())
+            .unwrap();
+        println!("✅")
+    }
+}
+
+#[test]
+fn test_input_descriptor_validation() -> Result<()> {
+    // Include the `input_descriptors_example.json` file in the `examples` directory.
+    let input_descriptors = include_str!(
+        "../tests/presentation-exchange/test/presentation-definition/input_descriptors_example.json"
+    );
```

Comment:
FYI: Submission Requirement groups is currently not handled in the validation flow.

Working on the `group` submission requirements in the validation flow to make this test work. Using the examples from the test examples git submodule.

Reply:
Submission requirement groups are now handled.